### PR TITLE
[go] Non-nullable fields

### PR DESCRIPTION
**Pull request by rvolosatovs**
Not quite sure if in `handlerclient` `app` is guaranteed to be non-nil. @htdvisser, is it? (I mean https://github.com/TheThingsNetwork/api/compare/feature/go-nullable?expand=1#diff-586f5b5b2a5a483a673498fcaf77d9ceR160). PR to ttn will be opened as soon as this is merged.

### CODE REVIEW

File: broker/getters.go [hunk 1]
```diff
@@ -7,17 +7,17 @@ import "github.com/TheThingsNetwork/ttn/core/types"
 
 func (m *DeviceActivationRequest) GetAppEUI() (eui types.AppEUI) {
 	if m != nil {
-		if m.AppEUI != nil {
-			return *m.AppEUI
+		if !m.AppEUI.IsEmpty() {
```
Comment (htdvisser) [inline broker/getters.go]: No need for this `if`. We're returning the `Empty` eui anyway. Similar changes below.

File: broker/message_marshaling.go [hunk 2]
```diff
@@ -65,13 +66,14 @@ func (m *DeviceActivationResponse) UnmarshalPayload() error {
...
 func (m *DeduplicatedUplinkMessage) UnmarshalPayload() error {
-	if m.GetMessage() == nil && m.GetProtocolMetadata() != nil && m.ProtocolMetadata.GetLoRaWAN() != nil {
+	if m.GetMessage() == nil && m.ProtocolMetadata.GetLoRaWAN() != nil {
 		msg, err := msgFromPayload(m.Payload)
 		if err != nil {
 			return err
 		}
 		m.Message = msg
-		if lorawan := m.GetProtocolMetadata().GetLoRaWAN(); lorawan != nil {
+		md := m.GetProtocolMetadata()
```
Comment (htdvisser) [inline broker/message_marshaling.go]: why add `md`?

File: discovery/getters.go [hunk 3]
```diff
@@ -7,8 +7,8 @@ import "github.com/TheThingsNetwork/ttn/core/types"
 
 func (m *GetByAppEUIRequest) GetAppEUI() (eui types.AppEUI) {
 	if m != nil {
-		if m.AppEUI != nil {
```
Comment (htdvisser) [inline discovery/getters.go]: no need for this `if` anymore

File: broker/getters.go (hunk 1 above)
Comment (rvolosatovs) [inline broker/getters.go]: Used `perl` magic for that, didn't think about the context too much

File: protocol/lorawan/getters.go [hunk 4]
```diff
@@ -7,35 +7,35 @@ import "github.com/TheThingsNetwork/ttn/core/types"
 
 func (m *DeviceIdentifier) GetAppEUI() (eui types.AppEUI) {
 	if m != nil {
-		if m.AppEUI != nil {
```
Comment (htdvisser) [inline protocol/lorawan/getters.go]: `if` is no longer needed, also below

File: router/getters.go [hunk 5]
```diff
@@ -7,17 +7,17 @@ import "github.com/TheThingsNetwork/ttn/core/types"
 
 func (m *DeviceActivationRequest) GetAppEUI() (eui types.AppEUI) {
 	if m != nil {
-		if m.AppEUI != nil {
```
Comment (htdvisser) [inline router/getters.go]: `if` is no longer needed, same below

File: router/message_marshaling.go [hunk 6]
```diff
@@ -22,13 +22,14 @@ func msgFromPayload(payload []byte) (*pb_protocol.Message, error) {
...
 func (m *UplinkMessage) UnmarshalPayload() error {
-	if m.GetMessage() == nil && m.GetProtocolMetadata() != nil && m.ProtocolMetadata.GetLoRaWAN() != nil {
+	if m.GetMessage() == nil && m.ProtocolMetadata.GetLoRaWAN() != nil {
 		msg, err := msgFromPayload(m.Payload)
 		if err != nil {
 			return err
 		}
 		m.Message = msg
-		if lorawan := m.GetProtocolMetadata().GetLoRaWAN(); lorawan != nil {
+		md := m.GetProtocolMetadata()
```
Comment (htdvisser) [inline router/message_marshaling.go]: why do we need `md`?

File: router/router.proto [hunk 7]
```diff
@@ -85,16 +85,16 @@ message GatewayStatusRequest {
...
 
 // message StatusRequest is used to request the status of this Router
 message StatusRequest {}
 
 // message Status is the response to the StatusRequest
 message Status {
-  api.SystemStats    system    = 1;
-  api.ComponentStats component = 2;
+  api.SystemStats    system    = 1 [(gogoproto.nullable) = false];
+  api.ComponentStats component = 2 [(gogoproto.nullable) = false];
```
Comment (htdvisser) [inline router/router.proto]: Can be nullable if gateway can't collect/send this information

File: broker/message_marshaling.go (hunk 2 above)
Comment (rvolosatovs) [inline broker/message_marshaling.go]: `GetLoRaWAN` has a pointer reciever :(

**Comment by htdvisser** [comment]
> Not quite sure if in handlerclient app is guaranteed to be non-nil.

`handlerclient` `DryUplink` and `DryDownlink` indeed have a non-nullable `App` field.

**Comment by rvolosatovs** [comment]
so the `*app` statements are fine?

**Comment by rvolosatovs** [comment]
@htdvisser

**Comment by htdvisser** [comment]
Yes, if no app is supplied, it should panic

**Comment by rvolosatovs** [comment]
@htdvisser

**Comment by htdvisser** [comment]
Github says no:

screen shot 2017-08-02 at 10 38 33

**Comment by rvolosatovs** [comment]
they were not there 7 days ago, when you approved the PR. I will make it work now

**Comment by rvolosatovs** [comment]
@htdvisser

**Comment by rvolosatovs** [comment]
@htdvisser

**Comment by htdvisser** [comment]
Travis CI is failing because `.travis.yml` is not in this branch yet. Merge is therefore OK

**Pull request merged**